examples/models/cubicmap.lisp

```diff
@@ -0,0 +1,67 @@
+(in-package #:cl-user)
+(defpackage claylib/examples/cubicmap
+  (:use :cl :claylib)
+  (:export :main))
+(in-package #:claylib/examples/cubicmap)
+
+(defparameter *scene*
+  (make-scene ((image-ass (make-image-asset (claylib/examples:claylib-path
+                                             "examples/models/resources/cubicmap.png")))
+               (texture (make-texture-asset (claylib/examples:claylib-path
+                                             "examples/models/resources/cubicmap_atlas.png"))))
+              ((cubicmap (let ((cubicmap (make-empty-texture
+                                          :rot 0
+                                          :tint +white+)))
+                           (load-texture-from-image (asset image-ass) :texture cubicmap)
+                           (setf (source cubicmap)
+                                 (make-simple-rec 0 0 (width cubicmap) (height cubicmap))
+                                 (dest cubicmap)
+                                 (make-simple-rec (- (get-screen-width) (* 4 (width cubicmap)) 20) 20
+                                                  (* 4 (width cubicmap)) (* 4 (height cubicmap))))
+                           cubicmap))
+               (model (let ((m (load-model-from-mesh (gen-mesh-cubicmap (asset image-ass)
+                                                                        (make-vector3 1 1 1)))))
+                        ;; Change the texture of the diffuse map of 0th material in the model
+                        (set-slot :texture
+                                  (elt (maps (elt (materials m) 0)) +material-map-diffuse+)
+                                  (asset texture))
+
+                        (setf (pos m) (make-vector3 -16 0 -8))
+                        ;; TODO Why doesn't this work?
+                        ;; (setf (x m) -16
+                        ;;       (y m) 0
+                        ;;       (z m) -8)
```

Comment:
Define "not work." Does it cause an error or just not get set?

Reply:
Lisp complains about %position being unbound. But looking at it again I realize exactly why. I stupidly thought `x`/`y`/`z` were C readers, but they are regular readers and the model does not start with a position when it's loaded from a mesh. I could give it a 3d 0-vector by default or just expect people to set the position like so `(setf (pos m) (make-vector3 -16 0 -8))` as I'm doing here. I think giving it a zero vector by default follow the principal of least surprise.

Reply:
Fixed by https://github.com/defun-games/claylib/pull/58/commits/803c8dbb6f1e4291aedd9650fdc2865ee1a36e83